moonstone/plot/counts.py

```diff
@@ -35,51 +43,207 @@ def plot_mean_distribution(
+                "title": f"{taxa_number} most abundant {taxa_level} - Total sum of abundances",
+                "xaxis_title": "Percentage Sample",
+                "yaxis_title": taxa_level.capitalize(),
+            }
+        }
+        if mean_taxa is not None:
+            default_plotting_options["layout"][
+                "title"
+            ] = "{} (mean among samples > {})".format(
+                default_plotting_options["layout"]["title"], mean_taxa
+            )
+        plotting_options = merge_dict(
+            kwargs.pop("plotting_options", {}), default_plotting_options
+        )
+        graph.plot_one_graph(
+            orientation="h", plotting_options=plotting_options, **kwargs
+        )
+
+    def _compute_abundances_for_n_taxa(
+        self, data_df: pd.DataFrame, taxa_number: int, taxa_level: str
+    ) -> pd.DataFrame:
+        """
+        Compute abundances for n (taxa_number) taxa with the rest grouped in Others.
+
+        Args:
+            data_df: Dataframe of abundances
+            taxa_number: Number of taxa to plot
+            taxa_level: Taxonomy level
+        """
+        df = data_df.groupby(taxa_level).sum()
+        top = (
+            df[~df.index.str.contains("(", regex=False)]
+            .sum(axis=1)
+            .sort_values(ascending=False)
+            .head(taxa_number)
+        )  # Filter out rows not classified to the species level (that contains '(')
+        top_and_other_df = df[df.index.get_level_values(taxa_level).isin(top.index)]
+        top_and_other_df = df.loc[
+            top.index
+        ]  # put top species in order from most abundant species across samples to least
+        top_and_other_df.loc["Others"] = 100 - top_and_other_df.sum()
+        return top_and_other_df
+
+    def _cluster_samples(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Reorder samples with clustering of given order list.
+
+        Args:
+            df: dataframe to reorder index
+        """
+        # Determine samples order using hierarchical clustering
+        Z = hierarchy.linkage(
+            distance.pdist(df.T),
+            method="single",
+            metric="euclidean",
+            optimal_ordering=False,
+        )
+        order = hierarchy.leaves_list(Z)
+        return df.iloc[:, order]
+
+    def plot_sample_composition_most_abundant_taxa(
+        self,
+        mean_taxa: float = None,
+        taxa_number: int = 20,
+        taxa_level: str = "species",
+        cluster_samples: bool = True,
+        samples_order: List[str] = None,
+        **kwargs,
+    ):
+        """
+        Plot taxa composition of samples for most abundant taxa.
+
+        Args:
+            mean_taxa: mean threshold to be kept for analysis
+            taxa_number: number of taxa to plot
+            taxa_level: Taxonomy level
+            cluster_samples: use clustering (skipped by samples_order)
+            samples_order: list of samples to force ordering
```

Comment:
Maybe just specify that it's only for the visualization but all samples are considered for the most abundant taxa computation

Reply:
Yes good idea